test/unit/ui_notifications/destroy_host_test.rb

```diff
@@ -7,20 +7,23 @@ class DestroyHostNotificationTest < ActiveSupport::TestCase
     assert blueprint
   end
 
-  test 'destorying discovered host should remove notification' do
+  test 'destroying discovered host should remove notification' do
     host = FactoryBot.create(:discovered_host)
+    ForemanDiscovery::UINotifications::NewHost.deliver!(host)
     assert_difference('blueprint.notifications.count', -1) do
       host.destroy
     end
   end
 
   test 'do not remove notification if other discovered hosts exists' do
     assert_difference('blueprint.notifications.count') do
-      FactoryBot.create(:discovered_host)
+      host = FactoryBot.create(:discovered_host)
```

Comment:
Since creating the host is not enough to have the notification (we need to parse facts too), I had to add `deliver!` so there are already some pre-existing notifications to delete